tests/lambda_endpoints/test_spice_indexer_lambda.py

```diff
@@ -0,0 +1,141 @@
+"""Tests for the SPICE indexer lambda."""
+
+import os
+from datetime import datetime
+
+from sds_data_manager.lambda_code.SDSCode.database import models
+from sds_data_manager.lambda_code.SDSCode.pipeline_lambdas import spice_indexer
+
+
+def put_local_file_in_bucket(s3_client, path_in_s3, path_local):
+    """Put the a local file into a test bucket, and return a mock event notification.
+
+    Parameters
+    ----------
+    s3_client
+        The s3 client to use
+    path_in_s3: str
+        The path to place the file in S3
+    path_local: str
+        The local path of the file to upload
+
+    Returns
+    -------
+    s3_event: dict
+        A dictionary mocking the s3 put event notification
+
+    """
+    with open(path_local, "rb") as f:
+        s3_client.put_object(
+            Bucket="test-data-bucket",
+            Key=path_in_s3,
+            Body=f,
+        )
+    event = {
+        "detail-type": "Object Created",
+        "source": "aws.s3",
+        "time": datetime.now().isoformat(),
+        "detail": {
+            "version": "0",
+            "bucket": {"name": "test-data-bucket"},
+            "object": {
+                "key": (path_in_s3),
+                "reason": "PutObject",
+            },
+        },
+    }
+    return event
+
+
+def test_s3_spice_files(session, s3_client, events_client):
+    """Test s3 event.
+
+    The following test mimics a leapsecond kernel being placed on the SDS,
+    followed by a spacecraft clock kernel, and then an attitude file.
+
+    The files are located in the "test_spice_files" directory.
+
+    """
+    current_path = os.path.dirname(os.path.abspath(__file__))
+
+    # Insert leapsecond spice kernel
+    leapsecond_event = put_local_file_in_bucket(
+        s3_client,
+        "spice/lsk/naif0012.tls",
+        current_path + "/test_spice_files/naif0012.tls",
+    )
+    spice_indexer.lambda_handler(leapsecond_event, None)
```

Comment:
Do we need to add a trigger for this case still? I figured this would automatically be handled and not need an explicit kickoff. Can we add a ticket for this if that is still work to be done?

Reply:
Do you have it set up so that the mocked infrastructure triggers the appropriate lambdas upon an upload? If so, I completely missed that!

Reply:
I am again not paying enough attention to this being a test file :(
It would be great if we did have this all set up like that! I don't think we do, but am not positive either.